toolset/setup/linux/frameworks/ulib.sh

```diff
@@ -28,6 +27,17 @@ if [ $? -ne 0 ]; then
    sudo apt-get install -y libcap2-bin
 fi
 
+# We need to install mongo-c-driver (we don't have a ubuntu package)
+RETCODE=$(fw_exists ${IROOT}/mongo-c-driver.installed)
+if [ "$RETCODE" != 0 ]; then
```

Comment:
This `RETCODE` business is unnecessary since you have the sentinel check above (`ulib.installed`). If the script gets here, then ulib is not installed and therefore the mongo-c-driver is not installed.